stable/concourse/values.yaml

```diff
@@ -609,3 +456,187 @@ rbac:
+      fdiTyQgJb+t8sSf+vDzPUs0hZjDaogzo2ff3TfxMLMDoIHnFItgfsdwn8QyygIZj
+      hC4pUQKBgQDqsxnkI6yXFE5gshnW7H8zqKNlzKd/dZEL6e+lRz4R3UY/KcEkRAfq
+      Yi3cwo9fE3U3kSmpl5MQwUjWm/BZ7JyueoY/4ndwaFPgc34IKsgJ0wau9pZiQAB1
+      DxpOSF+BR71Jx3sxvIdCODNTtm645j5yrZVnJAuMPofo5XFmunDoJA==
+      -----END RSA PRIVATE KEY-----
+
+    workerKeyPub: |-
+      ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC496FSYFcBAKgDtMsBAJiF/6/NxlXKP5UZecyEsedYuTt1GOgJTwaA1qZ1LmHsbfLDE68oDdiM4uvxfI4wtLhz57w3u0jOUxZ2JeF7SVwEf1nVqLn4Gh/f8GUNQGSyIp1zUD5Bx9fq0PAyQ47mt7Ufi84rcf8LKl7nzAIHTcdg2BvTkQN9bUGPaq/Pb1W2bKPAQy4OzXTSIyrAJ89TH2jFeaZfyxQFGbD9jVHH/yl0oiMrDeaRYgccE5II+KY7WoLjsBry/9Qf2ERELKTK4UeIGIqWci9lab1ti+GxFPPiC3krNFjo4jShV4eUs4cNIrjwNrxVaKPXmU6o7Y3Hpayx Concourse
+
+    ## Secrets for DB encryption
+    ##
+    # encryption:
+      # oldEncryptionKey:
+      # encryptionKey:
+
+    ## Concourse Basic Authentication Credentials
+    ## ref: https://concourse.ci/teams.html#authentication
+    basicAuth:
+      username: concourse
+      password: concourse
+
+    ## Secrets for GitHub OAuth.
+    ##
+    # githubAuth:
+      # clientId:
+      # clientSecret:
+
+    ## Secrets for GitLab OAuth.
+    ##
+    # gitlabAuth:
+      # clientId:
+      # clientSecret:
+
+    ## Secrets for generic OAuth.
+    ##
+    # genericOauth:
+      # clientId:
+      # clientSecret:
+
+  # postgresql:
+
+    ## If bringing your own PostgreSQL, set this to the full uri to use
+    ## e.g. postgres://concourse:changeme@my-postgres.com:5432/concourse?sslmode=disable
+    ##
+    # uri:
+
+  # credentialManager:
+
+    ## Secrets for Hashcorp Vault Credential Manager.
+    ##
+    # vault:
+
+      ## if the Vault server is using a self-signed certificate, provide the CA public key.
+      ## the value will be written to /concourse-vault/ca.cert
+      ##
+      # caCert: |-
+
+      ## initial periodic token issued for concourse
+      ## ref: https://www.vaultproject.io/docs/concepts/tokens.html#periodic-tokens
+      ##
+      # clientToken:
+
+      ## set role_id for [AppRole](https://www.vaultproject.io/docs/auth/approle.html) backend
+      ## make sure to also set authBackend to `approle`.
+      ##
+      # appRoleId:
+
+      ## set secret_id for [AppRole](https://www.vaultproject.io/docs/auth/approle.html) backend
+      ## make sure to also set authBackend to `approle`.
+      ##
+      # appRoleSecretId:
+
+      ## provide the client certificate for authenticating with the [TLS](https://www.vaultproject.io/docs/auth/cert.html) backend
+      ## the value will be written to /concourse-vault/client.cert
+      ## make sure to also set authBackend to `cert`
+      # clientCert: |-
+
+      ## provide the client key for authenticating with the [TLS](https://www.vaultproject.io/docs/auth/cert.html) backend
+      ## the value will be written to /concourse-vault/client.key
+      ## make sure to also set authBackend to `cert`
+      # clientKey: |-
```

Comment:
Some Vault deployments may have the TCP listener configured to require a client cert and client key alongside any other form of authentication so it is incorrect to make this a choice between the three. It must be client token or app role and/or client cert and client key. [Source](https://www.vaultproject.io/docs/configuration/listener/tcp.html#tls_require_and_verify_client_cert)

It is also possible to use one cert/key pair for the initial connection and another for the auth backend. (Untested but I'm pretty certain based on the documentation.)

An example use case could be having few shared client key pairs granting permission to talk with Vault at all (TCP listener) and a separate client key pair claiming the identity of the client. (TLS auth backend) This would allow for auditing of traffic with TLS decryption but securely prove the identity of the client in a way that could have a chain of trust subject to certificate revocation lists. (A common use case is for short lived ssh certs given to engineers by other engineers. Each engineer is an intermediate CA provided by Vault so provenance of credentials is easy to track.) 

Maybe it would be better to treat this like caCert instead by having it be independent from the auth settings?